### PR TITLE
Upgrade bootstrap

### PR DESCRIPTION
**Pull request by ConnorSheremeta**
- Upgraded bootstrap to version 4.3.1
- Updated css for new bootstrap version
- Updated views for new bootstrap version
- Updated facets_helper_behaviour.rb to get proper widths on facet-count spans for new bootstrap version
#1601 
Have a TODO in gemfile to reference ualbertalib's github fork and regenerate gemfile.lock once the PR's are merged. This was done for ease of testing.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,16 +8,18 @@ gem 'rails', '4.2.11.1'
...
 # Assets (CSS/JS) stuff
+gem 'bootstrap', '~> 4.3.1'
 gem 'jquery-rails'
 gem 'sassc-rails', '~> 2.1'
 gem 'uglifier', '>= 1.3.0'
 
 # blacklight stuff
-gem 'blacklight', '5.15.0'
-gem 'blacklight-marc', '~> 5.10.0'
+# TODO: make following 3 git gems link to ualbertalib's fork once PR's are approved and regenerate Gemfile.lock.
```
[the remark this answers is not in the document]
Comment (seanluyk) [reply Gemfile]: @pgwillia Thanks for these observations. I think I'd need to see them to know if they were an issue, but from your perspective are they improvements?

**Comment by pgwillia** [comment]
I pushed this branch from @ConnorSheremeta's fork to this repo and made [docker image](https://cloud.docker.com/u/ualbertalib/repository/registry-1.docker.io/ualbertalib/discovery/builds/bbec493d-db98-42c1-a026-0fb3f84dd3e3) from it.  This has been deployed on http://uat.library.ualberta.ca:3012 for UAT.

File: app/helpers/blacklight/facets_helper_behavior.rb [hunk 2]
```diff
@@ -144,7 +144,7 @@ def render_selected_facet_value(facet_solr_field, item)
       content_tag(:span, facet_display_value(facet_solr_field, item), class: 'selected') +
         # remove link
         link_to(
-          content_tag(:span, '', class: 'glyphicon glyphicon-remove') + content_tag(:span, '[remove]', class: 'sr-only'),
```
[the remark this answers is not in the document]
Comment (ConnorSheremeta) [reply app/helpers/blacklight/facets_helper_behavior.rb]: Blacklight didn't pick up a new icon font package so it's probably best to follow them

File: app/helpers/blacklight/facets_helper_behavior.rb [hunk 3]
```diff
@@ -161,7 +161,7 @@ def render_selected_facet_value(facet_solr_field, item)
   # @return [String]
   def render_facet_count(num, options = {})
     classes = (options[:classes] || []) << 'facet-count'
-    content_tag('span', t('blacklight.search.facets.count', number: number_with_delimiter(num)), class: classes)
+    content_tag('span', t('blacklight.search.facets.count', number: number_with_delimiter(num)), class: classes, style: 'width: ' + (number_with_delimiter(num).to_s.size + 1).to_s + 'ch;')
```
[the remark this answers is not in the document]
Comment (ConnorSheremeta) [reply app/helpers/blacklight/facets_helper_behavior.rb]: A width of auto just sets the cell to be half of the width of the table due to .facet-values having a display of table. This is a slightly modified version of Blacklight 7's solution which is here: https://github.com/projectblacklight/blacklight/blob/master/app/javascript/blacklight/facet_load.js

I noticed that there are some unintended consequences in how I've done it so my new commit just uses the above JS

**Comment by ConnorSheremeta** [comment]
Thanks for having this pushed to UAT tricia! I have addressed most of your check boxes and have left comments on the ones that I didn't

**Comment by seanluyk** [comment]
Thanks @ConnorSheremeta @pgwillia. This looks mostly good but before deploying it would be good to fix some of the weirdness on the catalog show page (text wrapping of field labels, tools menu styling) as well as those issues noticed on the search results page (e.g. small font for facet labels). Also wondering if @nnunn might have some time to take a closer look. If we can make any improvements along the way this is a good time to do so. I've always thought that font size could be improved anyway, for example

**Comment by ConnorSheremeta** [comment]
@seanluyk Catalog show page should now look proper on [UAT](http://uatsrv01.library.ualberta.ca:3012) with changes from #1678. The facet labels font hasn't really changed size (16.4pt vs 16.8pt) from this issue; If there should be changes to that font size, or other changes, they should probably be addressed in another issue as they are unrelated to the bootstrap upgrade.

**Comment by ConnorSheremeta** [comment]
I've increased the font size of facet-labels to their original size, nearly a 3pt increase (now on [UAT](http://uatsrv01.library.ualberta.ca:3012) .)

**Comment by ConnorSheremeta** [comment]
Just checking before I merge this, is this good to go @seanluyk ? I know you wanted @nnunn to have a look but as she's on vacation we've discussed that this should be merged and then she could have a look when she gets back.

**Comment by seanluyk** [comment]
@ConnorSheremeta this is looking really great, thanks so much for all your work! Since @nnunn is away let's merge this and then after the fact she can take a look and recommend any changes from her perspective. From my perspective, this is a huge improvement and a good time to be doing a UI refresh since it's quiet

**Pull request merged**